### PR TITLE
services/frontend-service: Add deployedBy and deployedAt to UI

### PR DESCRIPTION
**Pull request by gsandok**
Adds details on who deployed and when to the UI
Screenshot 2023-07-04 at 5 22 13 PM

### CODE REVIEW

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 1]
```diff
@@ -138,6 +138,24 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
         }
         return otherRelease?.sourceCommitId + ': ' + otherRelease?.sourceMessage;
     };
+    const getDeploymentAuthor = (): string => {
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:141]: it doesn't just return the author, maybe rename to `getDeploymentMetadata`?

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 2]
```diff
@@ -138,6 +138,24 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
...
+        if (!application) {
+            return ``;
+        }
+        if (!(application && application.version === release.version)) {
+            return '';
+        }
+        if (application.deploymentMetaData === null) {
+            return '';
+        }
+        var deployedUNIX = application.deploymentMetaData?.deployTime ?? '';
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:151]: please use `const`, or if not possible then `let`

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 3]
```diff
@@ -138,6 +138,24 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
...
+        if (deployedUNIX === '') {
+            return 'Deployed by: ' + application.deploymentMetaData?.deployAuthor;
+        }
+        var deployedDate = new Date(+deployedUNIX * 1000);
+        const returnString = 'Deployed by: ' + application.deploymentMetaData?.deployAuthor + ' on ' + deployedDate;
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:156]: please remove the `:`

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 4]
```diff
@@ -138,6 +138,24 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
...
+        if (!(application && application.version === release.version)) {
+            return '';
+        }
+        if (application.deploymentMetaData === null) {
+            return '';
+        }
+        var deployedUNIX = application.deploymentMetaData?.deployTime ?? '';
+        if (deployedUNIX === '') {
+            return 'Deployed by: ' + application.deploymentMetaData?.deployAuthor;
+        }
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:154]: e.g. like this:
```suggestion
        const deployedUNIX = application.deploymentMetaData?.deployTime;
        if (deployedUNIX) {
            return 'Deployed by: ' + application.deploymentMetaData?.deployAuthor;
        }
```

File: services/frontend-service/src/ui/components/ReleaseDialog/__snapshots__/ReleaseDialog.test.tsx.snap [hunk 5]
```diff
@@ -405,6 +410,11 @@ exports[`Release Dialog Renders the environment locks two envs release 1`] = `
                   >
                     cafe: the other commit message 2
                   </div>
+                  <div
+                    class="env-card-data"
+                  >
+                    Deployed by: undefined
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/__snapshots__/ReleaseDialog.test.tsx.snap:416]: We shouldn't print undefined, but instead "unknown".

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.test.tsx [hunk 6]
```diff
@@ -78,7 +78,7 @@ describe('Release Dialog', () => {
             ],
             expect_message: true,
             expect_queues: 0,
-            data_length: 1,
+            data_length: 2,
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.test.tsx:81]: for the screenshot, please add one where both time and name are set.

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx (hunk 4 above)
Comment (gsandok) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:154]: Without implicitly checking on line 151 if the time exists, line 155 will throw an error when trying to create deployedDate with the error: `'deployedUNIX' is possibly 'undefined'`

Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx]: I'm fine with how the code is written now.

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx [hunk 7]
```diff
@@ -138,6 +138,25 @@ export const EnvironmentListItem: React.FC<EnvironmentListItemProps> = ({
...
+        if (application.deploymentMetaData === null) {
+            return '';
+        }
+        const deployedBy = application.deploymentMetaData?.deployAuthor ?? 'unknown';
+        const deployedUNIX = application.deploymentMetaData?.deployTime ?? '';
+        if (deployedUNIX === '') {
+            return 'Deployed by ' + deployedBy;
+        }
+        const deployedDate = new Date(+deployedUNIX * 1000);
+        const returnString = 'Deployed by ' + deployedBy + ' on ' + deployedDate;
```
Comment (hannesg) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx:157]: For displaying dates we have a special component here: https://github.com/freiheit-com/kuberpult/blob/main/services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx#L52C8-L52C8

File: services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.test.tsx [hunk 8]
```diff
@@ -78,7 +78,47 @@ describe('Release Dialog', () => {
...
+            teamName: '',
+        },
+        {
+            name: 'normal release with deploymentMetadata set',
+            props: {
+                app: 'test1',
+                version: 2,
```
Comment (sven-urbanski-freiheit-com) [inline services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.test.tsx:88]: for the commit message (when merging), I would suggest:

> Display person who triggered deployment
> 
> Display the author and time of deployment in the release dialog.

Comment (gsandok) [reply services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.test.tsx:88]: Will change it when I merge, thanks !

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**